### DISCUSSION TITLE
Error in .rowNamesDF

### DISCUSSION THREAD

**Discussion by tatapinandhika** [teamalgoritma/community#142] [Q&A] [answered]
<img width="1440" alt="Screen Shot 2022-06-19 at 14 54 57" src="https://user-images.githubusercontent.com/101564223/174469520-3b21ab4d-dbcb-4f07-954b-25eaf35e64c9.png">

**Comment by onlyphantom** [comment, marked as answer]
Row names have to be unique. In `spotify$track_name` you have one or more values that appear more than once ("duplicate"), and that is the reason for the error message

> Error in `.rowNamesDF<-`(x, value = value) : 
  duplicate 'row.names' are not allowed

For example, try:

```
length(unique(spotify$track_name))
```
If you see that the length is 232,725 which is the `nrow(spotify)` that's when you know one or more values of track_name are duplicated. Fix them by giving them unique names, or merging them, and you'll be able to assign them to the row names of your dataframe again.

Good luck!
